### ISSUE TITLE
Typescript documentation is missing

### ISSUE THREAD

**Issue by jinshin1013** [wix/Detox#947]
First, thanks for the great testing framework!

I have seen @types/detox  done for types but could not find anywhere how I can run ts files with Detox config? Tried using ts-jest as test-runner but says it's not supported. Do I have to manually transform typescript files to Javascript file before test? I looked everywhere in the documentation but could not anything. 

* Detox: 9.0.1
* React Native: 0.56.1
* Node: 8.12.0
* Device: iPhone 7 Plus Simulator
* Xcode: 10
* macOS: 10.13.6

**Comment by solkaz** [comment]
I've created https://github.com/solkaz/ts-detox-example as an example of adding TypeScript to your Detox tests

The trickiest part is ensuring that Detox's typings don't trample over Jest's, specifically for your unit tests. I solved this by putting all of my end-to-end tests in an `e2e` directory at my project's root directory, then configured my main `tsconfig` to not include the typings in `e2e/node_modules/@types`.

Briefly, the steps are as follows:
- Add TypeScript, `ts-jest` and all necessary typings to your project
- Use `npx ts-jest config:init` to create `jest.config.js` (must be a `.js` file), then copy the `setupTestFrameworkScriptFile` over from `config.json`; then make this your `runner-config` in `package.json`.
- Convert your js files into ts files
